docs/taxonomy-revamp-2025.md

```diff
@@ -0,0 +1,93 @@
+  - The linter also complains about trailing whitespace, another common thing that the typical end user won't understand why everything is failing.
+
+From a code perspective,
+
+- We are already using JSON in the data mixing process in SDG[^10].
+- Docling also exports JSON as input and output[^11].
+- JSON is also much more friendly to UI work, which is a primary path we would like people to use.
+
+Overall, the `qna.yaml` file needs to have fewer knobs and fewer pitfalls.
+
+The process of writing question and answer sets also is more like writing reading comprehension sets from a standardized exam. It would be better to frame this hands-on part of the process as similar to the passage and question sets from English reading comprehension exams
+
+## Proposed solutions
+
+To fix the user experience when working with data, I propose the following ideas. In general, the basic idea is "Keep It Simple; Make It Tick."
+
+### Use a schema field rather than directory tree structure
+
+Drop the folder structure in favor of a schema field for submission type and even domain, if necessary. The schema field can be entered automatically via the UI through a user selecting `knowledge` or `skill`.
+
+### Streamline the schema
+
+Make `created_by`, `domain`, and `document_outline` optional fields. Enforce those fields for the upstream taxonomy though documentation, CI, and review rather than require them for everyone.
+
+### Switch to JSON and Markdown for the `qna.yaml` document
+
+Allow the user to use Markdown in a WYSIWYG experience, and then use a Markdown-to-JSON converter to handle the conversion to a code-friendly and machine-readable format.
+
+Markdown is very user-friendly, and converters handle a lot of the issues with encoding and special characters that happen in situations like working in other languages. We don't have to worry about a linter arguing about line length with the end user, and we wouldn't have to think about whether the user used tabs or spaces or forgot to strip whitespace at the end of a line.
+
+This would also make it a lot easier for the UI to work with contributions. JSON plays well with JavaScript overall without importing more libraries and creating dependency issues, and Python has a very good built-in for working with JSON files. Fewer dependencies means a smaller attack surface, as well.
+
+Users who decide to build it without needing the converter are likely familiar with JSON, and there are fewer pitfalls and less likelihood of tooling choices impacting meaning (e.g., where line breaks are for paragraph structures) as the JSON standard has not changed since 2017, and barely changed from the original standard.
+
+### Reframe the Q&A writing process as a reading comprehension process
```

Comment:
I think this is good for knowledge, I believe @abhi1092 is actually working on something along the lines of using natural concepts from reading comprehension for the Knowledge 1.5 pipeline. 

For skills training though it's probably a bit different since there we want the model to learn how to transform and permute different data.